scripts/datastore_upgrade.py

```diff
@@ -0,0 +1,422 @@
+    logging.info("AppScale is running, please shut it down and try again.")
+    sys.exit(1)
+
+def start_cassandra(status_dict, db_ips, keyname):
+  """ Creates a monit configuration file and prompts Monit to start Cassandra.
+  Args:
+    status_dict: A dictionary to record the status of the executed process.
+    db_ips: A list of database node IPs to start Cassandra on.
+    keyname: A string containing the deployment's keyname.
+  """
+  logging.info("Starting Cassandra...")
+  for ip in db_ips:
+    start_service_cmd = START_SERVICE_SCRIPT + CASSANDRA_WATCH_NAME
+    cmd_status = utils.ssh(ip, keyname, start_service_cmd)
+    start_cassandra_ip = START_CASSANDRA + ":" + ip
+    if not cmd_status == 0:
+      logging.error("Monit was unable to start Cassandra.")
+      status_dict[start_cassandra_ip] = FAILURE
+      sys.exit(1)
+    logging.info("Successfully started Cassandra.")
+    status_dict[start_cassandra_ip] = SUCCESS
+
+def start_zookeeper(status_dict, zk_ips, keyname):
+  """ Creates a monit configuration file and prompts Monit to start Cassandra.
+    Args:
+      status_dict: A dictionary to record the status of the executed process.
+      zk_ips: A list of zookeeper node IPs to start ZooKeeper on.
+      keyname: A string containing the deployment's keyname.
+    """
+  logging.info("Starting ZooKeeper...")
+  for ip in zk_ips:
+    start_service_cmd = START_SERVICE_SCRIPT + ZK_WATCH_NAME
+    cmd_status = utils.ssh(ip, keyname, start_service_cmd)
+    start_zookeeper_ip = START_ZOOKEEPER + ":" + ip
+    if not cmd_status == 0:
+      logging.error("Monit was unable to start ZooKeeper.")
+      status_dict[start_zookeeper_ip] = FAILURE
+      sys.exit(1)
+    logging.info("Successfully started ZooKeeper.")
+    status_dict[start_zookeeper_ip] = SUCCESS
+
+def get_datastore():
+  """ Returns a reference to the batch datastore interface. Validates where
+  the <datastore>_interface.py is and adds that path to the system path.
+  """
+  db_info = appscale_info.get_db_info()
+  db_type = db_info[':table']
+  datastore_batch = appscale_datastore_batch.DatastoreFactory.getDatastore(db_type)
+  return datastore_batch
+
+def get_datastore_distributed(datastore, zk_location_ips):
+  """ Returns a reference to the distributed datastore.
+  Args:
+    datastore: A reference to the batch datastore interface.
+    zk_location_ips: A list of ZooKeeper location ips.
+  """
+  zookeeper_locations = get_zk_locations_string(zk_location_ips)
+  zookeeper = zk.ZKTransaction(host=zookeeper_locations)
+  datastore_distributed = datastore_server.DatastoreDistributed(datastore, zookeeper=zookeeper)
+  return datastore_distributed, zookeeper
+
+def get_zk_locations_string(zk_location_ips):
+  """ Generates a ZooKeeper IP locations string.
+  Args:
+    zk_location_ips: A list of ZooKeeper IP locations.
+  Returns:
+    A string of ZooKeeper IP locations.
+  """
+  return (ZK_PORT_WITH_COLON + ",").join(zk_location_ips) + ZK_PORT_WITH_COLON
```

Comment:
You can change this into something like: 
`(':' + ZK_PORT + ",").join(zk_location_ips) + ZK_PORT`